
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>SNV &#8212; baseq 0.01 documentation</title>
    <link rel="stylesheet" href="../_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <script type="text/javascript" src="../_static/documentation_options.js"></script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="RNA-Seq" href="../RNA/index.html" />
    <link rel="prev" title="CNV" href="CNV.html" />
   
  <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="snv">
<h1>SNV<a class="headerlink" href="#snv" title="Permalink to this headline">¶</a></h1>
<p>SNV analysis functions.</p>
<div class="section" id="quality-control">
<h2>Quality Control<a class="headerlink" href="#quality-control" title="Permalink to this headline">¶</a></h2>
<div class="section" id="module-baseq.snv.qc.enrich">
<span id="enrichment"></span><h3>Enrichment<a class="headerlink" href="#module-baseq.snv.qc.enrich" title="Permalink to this headline">¶</a></h3>
<dl class="function">
<dt id="baseq.snv.qc.enrich.quality_control">
<code class="descclassname">baseq.snv.qc.enrich.</code><code class="descname">quality_control</code><span class="sig-paren">(</span><em>samplename</em>, <em>bampath</em>, <em>intervals</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/baseq/snv/qc/enrich.html#quality_control"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#baseq.snv.qc.enrich.quality_control" title="Permalink to this definition">¶</a></dt>
<dd><p>Check the coverage depth and enrichment quality.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">quality_control</span><span class="p">(</span><span class="s2">&quot;sample01&quot;</span><span class="p">,</span> <span class="s2">&quot;xx.bam&quot;</span><span class="p">,</span> <span class="s2">&quot;panel.bed&quot;</span><span class="p">)</span>
</pre></div>
</div>
<dl class="docutils">
<dt>Return:</dt>
<dd>Sample/Total/Mapped/Map_Ratio/Dup_ratio/PCT_10X/PCT_30X/…</dd>
</dl>
</dd></dl>

</div>
</div>
<div class="section" id="vcf">
<h2>VCF<a class="headerlink" href="#vcf" title="Permalink to this headline">¶</a></h2>
<div class="section" id="module-baseq.snv.vcf.GATK">
<span id="vcf-stats-and-filter"></span><h3>VCF stats and Filter<a class="headerlink" href="#module-baseq.snv.vcf.GATK" title="Permalink to this headline">¶</a></h3>
<dl class="function">
<dt id="baseq.snv.vcf.GATK.vcf_stats">
<code class="descclassname">baseq.snv.vcf.GATK.</code><code class="descname">vcf_stats</code><span class="sig-paren">(</span><em>sample</em>, <em>vcfpath</em>, <em>min_depth=50</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/baseq/snv/vcf/GATK.html#vcf_stats"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#baseq.snv.vcf.GATK.vcf_stats" title="Permalink to this definition">¶</a></dt>
<dd><p>Stats on the VCF from GATK</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">vcf_stats</span><span class="p">(</span><span class="s2">&quot;sample1&quot;</span><span class="p">,</span> <span class="s2">&quot;path/to/vcf&quot;</span><span class="p">,</span> <span class="n">min_depth</span><span class="o">=</span><span class="mi">30</span><span class="p">)</span>
</pre></div>
</div>
<dl class="docutils">
<dt>Return:</dt>
<dd>A dict/json containing:
Samplename/counts/mean_depth/GT_01/GT_11/MAF
MAF is minor allel frequency.</dd>
</dl>
</dd></dl>

</div>
</div>
<div class="section" id="gatk">
<h2>GATK<a class="headerlink" href="#gatk" title="Permalink to this headline">¶</a></h2>
<div class="section" id="module-baseq.snv.gatk">
<span id="run-gatk"></span><h3>run GATK<a class="headerlink" href="#module-baseq.snv.gatk" title="Permalink to this headline">¶</a></h3>
<dl class="function">
<dt id="baseq.snv.gatk.alignment">
<code class="descclassname">baseq.snv.gatk.</code><code class="descname">alignment</code><span class="sig-paren">(</span><em>fq1</em>, <em>fq2</em>, <em>sample</em>, <em>genome</em>, <em>thread=8</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/baseq/snv/gatk.html#alignment"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#baseq.snv.gatk.alignment" title="Permalink to this definition">¶</a></dt>
<dd><p>Map low-divergent sequences against reference genome using BWA.
Add ReadGroup(more details about <a class="reference external" href="https://software.broadinstitute.org/gatk/documentation/article.php?id=6472">ReadGroup</a> )to bamfile using the input sample name.
Outfile is in BAM format and indexed for the downstream analysis.</p>
<p>Usage:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">baseq</span><span class="o">-</span><span class="n">SNV</span> <span class="n">run_bwa</span> <span class="o">-</span><span class="mi">1</span> <span class="n">Reads</span><span class="o">.</span><span class="mf">1.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span> <span class="o">-</span><span class="mi">2</span> <span class="n">Read</span><span class="o">.</span><span class="mf">2.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span> <span class="o">-</span><span class="n">g</span> <span class="n">hg38</span> <span class="o">-</span><span class="n">n</span> <span class="n">Test</span>
</pre></div>
</div>
<p>Return:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">Test</span><span class="o">.</span><span class="n">bam</span>
<span class="n">Test</span><span class="o">.</span><span class="n">bam</span><span class="o">.</span><span class="n">bai</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="baseq.snv.gatk.bqsr">
<code class="descclassname">baseq.snv.gatk.</code><code class="descname">bqsr</code><span class="sig-paren">(</span><em>markedbam</em>, <em>bqsrbam</em>, <em>genome</em>, <em>disable_dup_filter=False</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/baseq/snv/gatk.html#bqsr"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#baseq.snv.gatk.bqsr" title="Permalink to this definition">¶</a></dt>
<dd><p>Run <a class="reference external" href="https://gatkforums.broadinstitute.org/gatk/discussion/44/base-quality-score-recalibration-bqsr">BQSR</a>. This function performs the two-steps process called base quality score recalibration. the first
ster generates a recalibration table based on various covariates which is recruited to the second step to
correct the systematic bias in input BAM file. More details about <a class="reference external" href="https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_hellbender_tools_walkers_bqsr_BaseRecalibrator.php">BaseRecalibrator</a> and <a class="reference external" href="https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_hellbender_tools_walkers_bqsr_ApplyBQSR.php">ApplyBQSR</a> .</p>
<p>Usage:</p>
<ul>
<li><p class="first">Default mode filters duplicate reads (reads with “markduplicate” tags) before applying BQSR</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">baseq</span><span class="o">-</span><span class="n">SNV</span> <span class="n">run_bqsr</span> <span class="o">-</span><span class="n">m</span> <span class="n">Test</span><span class="o">.</span><span class="n">marked</span><span class="o">.</span><span class="n">bam</span> <span class="o">-</span><span class="n">g</span> <span class="n">hg38</span> <span class="o">-</span><span class="n">q</span> <span class="n">Test</span><span class="o">.</span><span class="n">marked</span><span class="o">.</span><span class="n">bqsr</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
</li>
<li><p class="first">Disable reads filter before analysis.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">baseq</span><span class="o">-</span><span class="n">SNV</span> <span class="n">run_bqsr</span> <span class="o">-</span><span class="n">m</span> <span class="n">Test</span><span class="o">.</span><span class="n">marked</span><span class="o">.</span><span class="n">bam</span> <span class="o">-</span><span class="n">g</span> <span class="n">hg38</span> <span class="o">-</span><span class="n">q</span> <span class="n">Test</span><span class="o">.</span><span class="n">marked</span><span class="o">.</span><span class="n">bqsr</span><span class="o">.</span><span class="n">bam</span> <span class="o">-</span><span class="n">f</span> <span class="n">Yes</span>
</pre></div>
</div>
</li>
</ul>
<p>Return:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">Test</span><span class="o">.</span><span class="n">marked</span><span class="o">.</span><span class="n">bam</span><span class="o">.</span><span class="n">table</span>
<span class="n">Test</span><span class="o">.</span><span class="n">marked</span><span class="o">.</span><span class="n">bqsr</span><span class="o">.</span><span class="n">bai</span>
<span class="n">Test</span><span class="o">.</span><span class="n">marked</span><span class="o">.</span><span class="n">bqsr</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="baseq.snv.gatk.create_pon">
<code class="descclassname">baseq.snv.gatk.</code><code class="descname">create_pon</code><span class="sig-paren">(</span><em>genome</em>, <em>list</em>, <em>path</em>, <em>interval</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/baseq/snv/gatk.html#create_pon"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#baseq.snv.gatk.create_pon" title="Permalink to this definition">¶</a></dt>
<dd><p>Create_pon function helps you create PoN(panel of normals) file necessary for mutect2 calling. The PoN captures
common artifactual and germline variant sites. Mutect2 then uses the PoN to filter variants at the site-level.</p>
<p>Example of samples list (tab delimited):</p>
<ul>
<li><p class="first">Content of columns are normal sample name, normal BAM file, tumor sample name, tumor BAM file(order cannot be distruped).
Absoulte path of all BAM files should be added if directory of BAM files and analysis directory are different.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">N504</span>    <span class="n">N504_marked_bqsr</span><span class="o">.</span><span class="n">bam</span>   <span class="n">T504</span>    <span class="n">T504_marked_bqsr</span><span class="o">.</span><span class="n">bam</span>
<span class="n">N505</span>    <span class="n">N505_marked_bqsr</span><span class="o">.</span><span class="n">bam</span>   <span class="n">T505</span>    <span class="n">T505_marked_bqsr</span><span class="o">.</span><span class="n">bam</span>
<span class="n">N506</span>    <span class="n">N506_marked_bqsr</span><span class="o">.</span><span class="n">bam</span>   <span class="n">T506</span>    <span class="n">T506_marked_bqsr</span><span class="o">.</span><span class="n">bam</span>
<span class="n">N509</span>    <span class="n">N509_marked_bqsr</span><span class="o">.</span><span class="n">bam</span>   <span class="n">T509</span>    <span class="n">T509_marked_bqsr</span><span class="o">.</span><span class="n">bam</span>
<span class="n">N510</span>    <span class="n">N510_marked_bqsr</span><span class="o">.</span><span class="n">bam</span>   <span class="n">T510</span>    <span class="n">T510_marked_bqsr</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
</li>
</ul>
<p>Usage:</p>
<ul>
<li><p class="first">Interval list defines genomic regions where analysis is restricted. Introduction of interval list format and its function, please see <a class="reference external" href="https://software.broadinstitute.org/gatk/documentation/article?id=11009">here</a>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># designated a intervals.list</span>
<span class="n">baseq</span><span class="o">-</span><span class="n">SNV</span> <span class="n">create_pon</span> <span class="o">-</span><span class="n">g</span> <span class="n">hg37</span> <span class="o">-</span><span class="n">l</span> <span class="n">sample_list</span><span class="o">.</span><span class="n">txt</span> <span class="o">-</span><span class="n">p</span> <span class="o">./</span> <span class="o">-</span><span class="n">L</span> <span class="n">interval</span><span class="o">.</span><span class="n">list</span>
<span class="c1"># Using the dafalut intervals.list</span>
<span class="n">baseq</span><span class="o">-</span><span class="n">SNV</span> <span class="n">create_pon</span> <span class="o">-</span><span class="n">g</span> <span class="n">hg37</span> <span class="o">-</span><span class="n">l</span> <span class="n">sample_list</span><span class="o">.</span><span class="n">txt</span> <span class="o">-</span><span class="n">p</span> <span class="o">./</span>
</pre></div>
</div>
</li>
</ul>
</dd></dl>

<dl class="function">
<dt id="baseq.snv.gatk.mutect2">
<code class="descclassname">baseq.snv.gatk.</code><code class="descname">mutect2</code><span class="sig-paren">(</span><em>genome</em>, <em>normalname</em>, <em>normalbam</em>, <em>tumorname</em>, <em>tumorbam</em>, <em>vcffile</em>, <em>pon</em>, <em>germline</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/baseq/snv/gatk.html#mutect2"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#baseq.snv.gatk.mutect2" title="Permalink to this definition">¶</a></dt>
<dd><p>Mutect2 is aim to call somatic SNVs and indels via local assembly of haplotypes. This function requires both
tumor BAM file and its matched normal BAM file. tumorname and normalname should be consistent with the ReadGroup(ID) of tumor
BAM file and normal BAM file respectively. PoN is refer to panel of normals callset(more infomation about PoN and how to
create it, please see <a class="reference external" href="https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_hellbender_tools_walkers_mutect_CreateSomaticPanelOfNormals.php">PoN</a> ). Germline resource, also in VCF format, is used to annotate variant alleles. Download <a class="reference external" href="https://software.broadinstitute.org/gatk/download/bundle">germline</a> resource.</p>
<p>Usage:</p>
<ul>
<li><p class="first">Simplified Mutect2 command line</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># single sample</span>
<span class="n">baseq</span><span class="o">-</span><span class="n">SNV</span> <span class="n">run_mutect2</span> <span class="o">-</span><span class="n">g</span> <span class="n">hg37</span> <span class="o">-</span><span class="n">n</span> <span class="n">normal</span> <span class="o">-</span><span class="n">N</span> <span class="n">normal_marked_bqsr</span><span class="o">.</span><span class="n">bam</span> \
                              <span class="o">-</span><span class="n">t</span> <span class="n">tumor</span> <span class="o">-</span><span class="n">T</span> <span class="n">tumor_marked_bqsr</span><span class="o">.</span><span class="n">bam</span> <span class="o">-</span><span class="n">o</span> <span class="o">./</span>
<span class="c1"># multiple samples</span>
<span class="n">baseq</span><span class="o">-</span><span class="n">SNV</span> <span class="n">run_mutect2</span> <span class="o">-</span><span class="n">g</span> <span class="n">hg37</span> <span class="o">-</span><span class="n">l</span> <span class="n">sample_list</span><span class="o">.</span><span class="n">txt</span> <span class="o">-</span><span class="n">o</span> <span class="o">./</span>
</pre></div>
</div>
</li>
<li><p class="first">Specify PoN(panels of normals) VCF file and germline VCF file
Default germline VCF file comes form GATK resource bundle and is recruited if germline isn’t designated.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># single sample</span>
<span class="n">baseq</span><span class="o">-</span><span class="n">SNV</span> <span class="n">run_mutect2</span> <span class="o">-</span><span class="n">g</span> <span class="n">hg37</span> <span class="o">-</span><span class="n">n</span> <span class="n">normal</span> <span class="o">-</span><span class="n">N</span> <span class="n">normal_marked_bqsr</span><span class="o">.</span><span class="n">bam</span> \
                              <span class="o">-</span><span class="n">t</span> <span class="n">tumor</span> <span class="o">-</span><span class="n">T</span> <span class="n">tumor_marked_bqsr</span><span class="o">.</span><span class="n">bam</span> <span class="o">-</span><span class="n">o</span> <span class="o">./</span> \
                              <span class="o">-</span><span class="n">p</span> <span class="n">pon</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">-</span><span class="n">G</span> <span class="n">af</span><span class="o">-</span><span class="n">only</span><span class="o">-</span><span class="n">gnomad</span><span class="o">.</span><span class="n">raw</span><span class="o">.</span><span class="n">sites</span><span class="o">.</span><span class="n">b37</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>
<span class="c1"># multiple samples</span>
<span class="n">baseq</span><span class="o">-</span><span class="n">SNV</span> <span class="n">run_mutect2</span> <span class="o">-</span><span class="n">g</span> <span class="n">hg37</span> <span class="o">-</span><span class="n">l</span> <span class="n">sample_list</span><span class="o">.</span><span class="n">txt</span> <span class="o">-</span><span class="n">o</span> <span class="o">./</span> \
                              <span class="o">-</span><span class="n">p</span> <span class="n">pon</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">-</span><span class="n">G</span> <span class="n">af</span><span class="o">-</span><span class="n">only</span><span class="o">-</span><span class="n">gnomad</span><span class="o">.</span><span class="n">raw</span><span class="o">.</span><span class="n">sites</span><span class="o">.</span><span class="n">b37</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
</li>
</ul>
</dd></dl>

<dl class="function">
<dt id="baseq.snv.gatk.run_callvar">
<code class="descclassname">baseq.snv.gatk.</code><code class="descname">run_callvar</code><span class="sig-paren">(</span><em>bqsrbam</em>, <em>rawvcf</em>, <em>genome</em>, <em>disable_dup_filter=False</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/baseq/snv/gatk.html#run_callvar"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#baseq.snv.gatk.run_callvar" title="Permalink to this definition">¶</a></dt>
<dd><p>Call germline SNPs and indels via local re-assembly of haplotypes. BAM file recalbrated by BQSR do recommand as
input BAM file and this functin only run the single sample genotypeVCF calling. More details see also
<a class="reference external" href="https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_hellbender_tools_walkers_haplotypecaller_HaplotypeCaller.php">HaplotypeCaller</a></p>
<p>Usage:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">baseq</span><span class="o">-</span><span class="n">SNV</span> <span class="n">run_callvar</span> <span class="o">-</span><span class="n">q</span> <span class="n">Test</span><span class="o">.</span><span class="n">marked</span><span class="o">.</span><span class="n">bqsr</span><span class="o">.</span><span class="n">bam</span> <span class="o">-</span><span class="n">r</span> <span class="n">Test</span><span class="o">.</span><span class="n">raw</span><span class="o">.</span><span class="n">indel</span><span class="o">.</span><span class="n">snp</span><span class="o">.</span><span class="n">vcf</span> <span class="o">-</span><span class="n">g</span> <span class="n">hg38</span>
</pre></div>
</div>
<p>Return:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">Test</span><span class="o">.</span><span class="n">raw</span><span class="o">.</span><span class="n">indel</span><span class="o">.</span><span class="n">snp</span><span class="o">.</span><span class="n">vcf</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="baseq.snv.gatk.run_markdup">
<code class="descclassname">baseq.snv.gatk.</code><code class="descname">run_markdup</code><span class="sig-paren">(</span><em>bamfile</em>, <em>markedbam</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/baseq/snv/gatk.html#run_markdup"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#baseq.snv.gatk.run_markdup" title="Permalink to this definition">¶</a></dt>
<dd><p>Run MarkDuplicates of Picard. this function tags duplicate reads with “markduplicate” in BAM file.
See also <a class="reference external" href="https://software.broadinstitute.org/gatk/documentation/tooldocs/current/picard_sam_markduplicates_MarkDuplicates.php">MarkDuplicates</a> in GATK.</p>
<p>Usage:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">baseq</span><span class="o">-</span><span class="n">SNV</span> <span class="n">run_markdup</span> <span class="o">-</span><span class="n">b</span> <span class="n">Test</span><span class="o">.</span><span class="n">bam</span> <span class="o">-</span><span class="n">m</span> <span class="n">Test</span><span class="o">.</span><span class="n">marked</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
<p>Return:
metrics file indicates the numbers of duplicates for both single- and paired-end reads</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">Test</span><span class="o">.</span><span class="n">marked</span><span class="o">.</span><span class="n">bam</span>
<span class="n">Test</span><span class="o">.</span><span class="n">marked</span><span class="o">.</span><span class="n">bam</span><span class="o">.</span><span class="n">bai</span>
<span class="n">Test</span><span class="o">.</span><span class="n">marked</span><span class="o">.</span><span class="n">bam</span><span class="o">.</span><span class="n">metrics</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="baseq.snv.gatk.selectvar">
<code class="descclassname">baseq.snv.gatk.</code><code class="descname">selectvar</code><span class="sig-paren">(</span><em>rawvcf</em>, <em>selectvcf</em>, <em>filtervcf</em>, <em>genome</em>, <em>run=True</em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/baseq/snv/gatk.html#selectvar"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#baseq.snv.gatk.selectvar" title="Permalink to this definition">¶</a></dt>
<dd><p>This function selects SNPs from a VCF file which is usually the output file of
HaplotypeCaller. Then, all SNPs are filtered by certain criteria based on INFO and/or FORMAT annotations.
Criteria used here is “QD &lt; 2.0 || FS &gt; 60.0 || MQ &lt; 40.0”.
More details about <a class="reference external" href="https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_hellbender_tools_walkers_variantutils_SelectVariants.php">SelectVariants</a> and <a class="reference external" href="https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_hellbender_tools_walkers_filters_VariantFiltration.php">VariantFiltration</a></p>
<p>Usage:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">baseq</span><span class="o">-</span><span class="n">SNV</span> <span class="n">run_selectvar</span> <span class="o">-</span><span class="n">r</span> <span class="n">Test</span><span class="o">.</span><span class="n">raw</span><span class="o">.</span><span class="n">indel</span><span class="o">.</span><span class="n">snp</span><span class="o">.</span><span class="n">vcf</span> <span class="o">-</span><span class="n">s</span> <span class="n">Test</span><span class="o">.</span><span class="n">raw</span><span class="o">.</span><span class="n">snp</span><span class="o">.</span><span class="n">vcf</span> <span class="o">-</span><span class="n">f</span> <span class="n">Test</span><span class="o">.</span><span class="n">filtered</span><span class="o">.</span><span class="n">snp</span><span class="o">.</span><span class="n">vcf</span> <span class="o">-</span><span class="n">g</span> <span class="n">hg38</span>
</pre></div>
</div>
<p>Return:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">Test</span><span class="o">.</span><span class="n">raw</span><span class="o">.</span><span class="n">snp</span><span class="o">.</span><span class="n">vcf</span>
<span class="n">Test</span><span class="o">.</span><span class="n">filtered</span><span class="o">.</span><span class="n">snp</span><span class="o">.</span><span class="n">vcf</span>
</pre></div>
</div>
</dd></dl>

</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="../index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">SNV</a><ul>
<li><a class="reference internal" href="#quality-control">Quality Control</a><ul>
<li><a class="reference internal" href="#module-baseq.snv.qc.enrich">Enrichment</a></li>
</ul>
</li>
<li><a class="reference internal" href="#vcf">VCF</a><ul>
<li><a class="reference internal" href="#module-baseq.snv.vcf.GATK">VCF stats and Filter</a></li>
</ul>
</li>
<li><a class="reference internal" href="#gatk">GATK</a><ul>
<li><a class="reference internal" href="#module-baseq.snv.gatk">run GATK</a></li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../index.html">Documentation overview</a><ul>
      <li>Previous: <a href="CNV.html" title="previous chapter">CNV</a></li>
      <li>Next: <a href="../RNA/index.html" title="next chapter">RNA-Seq</a></li>
  </ul></li>
</ul>
</div>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="../_sources/baseq/SNV.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2018, Xiannian Zhang.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.7.4</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.10</a>
      
      |
      <a href="../_sources/baseq/SNV.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>